### ISSUE TITLE
sqlfluff cannot parse array slicing with missing bound in PostgreSQL

### ISSUE THREAD

**Issue by mummyhen** [sqlfluff/sqlfluff#3842]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```sql
SELECT 
   my_array[2:]
FROM my_table
```

throws 
> L:  47 | P:   4 |  PRS | Line 47, Position 4: Found unparsable section: '[2:]'
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly? 

but 
```sql
SELECT 
   my_array[2:3]
FROM my_table
```
works correctly

### Expected Behaviour

PostgreSQL supports array slicing where the lower and/or upper bound has been omitted: https://www.postgresql.org/docs/14/arrays.html

### Observed Behaviour

> L:  47 | P:   4 |  PRS | Line 47, Position 4: Found unparsable section: '[2:]'
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly? 

### How to reproduce

Run linter on
```sql
WITH

my_table AS (
	SELECT '{2,3}'::INT[] AS my_array
)

SELECT
	my_array[2:]
FROM my_table
```

### Dialect

PostgreSQL + dbt

### Version

1.3.0

### Configuration

.sqlfluff

```
[sqlfluff]
templater = dbt
dialect = postgres

# rules = None
exclude_rules = L034, L031

[sqlfluff:indentation]
indented_joins = True

[sqlfluff:rules:L004]  # Indent
indent_unit = tab

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = upper

[sqlfluff:rules:L014]  # Unquoted identifiers
extended_capitalisation_policy = lower

[sqlfluff:rules:L059]  # Unquoted identifiers
ignore_words = type,comment,name, label
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Thanks for posting this, so that I'm clear - does this notation have the same effect as _slicing_ in a python context: i.e. give me a subsection of the array, starting at the 3rd element?

**Comment by mummyhen** [comment]
Yes it's exactly the same as slicing in Python. You can omit the lower bound, upper bound or both
